src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.WorkspaceHostState.cs

```diff
@@ -90,6 +105,14 @@ internal void StartPushingToWorkspaceAndNotifyOfOpenDocuments(IEnumerable<Abstra
 
                 var projectInfos = inOrderToPush.Select(p => p.CreateProjectInfoForCurrentState()).ToImmutableArray();
 
+                // Enable projects to start pushing changes to workspace hosts, even before we add the solution/project to the host.
```

Comment:
The issue is that between the point we capture the project info for current state above and the point where we start pushing to workspace hosts, CPS may send new events on the AbstractProject on a background thread, and these won't get pushed over to the workspace hosts as we didn't set the `_pushingChangesToWorkspaceHost` flag on the AbstractProject. If we set this flag upfront, then such background thread events with ensure that notification to workspace host is queued up on foreground scheduled tasks.